semver/semver.go

```diff
@@ -224,6 +224,12 @@ func recursivePreReleaseCompare(versionA []string, versionB []string) int {
 		bInt = true
 	}
 
+	if aInt && !bInt {
```

Comment:
For posterity, would you mind adding a comment here?
`// Numeric identifiers always have lower precedence than non-numeric identifiers.`